src/Microsoft.DotNet.VersionTools/Automation/GitHubHttpClient.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.VersionTools.Automation
+{
+    public class GitHubHttpClient : HttpClient
+    {
+        private GitHubAuth _auth;
+
+        public GitHubHttpClient(GitHubAuth auth)
+        {
+            DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json");
+            DefaultRequestHeaders.Add("Authorization", $"token {auth.AuthToken}");
+            DefaultRequestHeaders.Add("User-Agent", auth.User);
+            _auth = auth;
+        }
+
+        public async Task<HttpResponseMessage> PutGitHubFileAsync(
+            string fileUrl,
+            string commitMessage,
+            string newFileContents)
+        {
+            Trace.TraceInformation($"Getting the 'sha' of the current contents of file '{fileUrl}'");
+
+            string currentFile = await GetStringAsync(fileUrl);
+            string currentSha = JObject.Parse(currentFile)["sha"].ToString();
+
+            Trace.TraceInformation($"Got 'sha' value of '{currentSha}'");
+
+            Trace.TraceInformation($"Request to update file '{fileUrl}' contents to:");
+            Trace.TraceInformation(newFileContents);
+
+            string updateFileBody = JsonConvert.SerializeObject(new
+            {
+                message = commitMessage,
+                committer = new
+                {
+                    name = _auth.User,
+                    email = _auth.Email
+                },
+                content = ToBase64(newFileContents),
+                sha = currentSha
+            }, Formatting.Indented);
+
+            var bodyContent = new StringContent(updateFileBody);
+            using (HttpResponseMessage response = await PutAsync(fileUrl, bodyContent))
+            {
+                response.EnsureSuccessStatusCode();
+                Trace.TraceInformation("Updated the file successfully.");
+
+                return response;
```

Comment:
Fixed here too by changing to `Task` and removing return.

---
In reply to: [69951690](https://github.com/dotnet/buildtools/pull/863#discussion_r69951690) [](ancestors = 69951690)